app/src/main/java/com/android/unio/ui/association/AssociationSmall.kt

```diff
@@ -23,13 +30,28 @@ import com.android.unio.ui.theme.AppTypography
  * @param onClick Callback when the association is clicked.
  */
 @Composable
-fun AssociationSmall(association: Association, onClick: () -> Unit) {
+fun AssociationSmall(association: Association, userUid: String, onClick: () -> Unit) {
+  val userRole = getUserRoleInAssociation(association, userUid)
   TextButton(onClick = onClick, modifier = Modifier.fillMaxWidth()) {
     Row(
         verticalAlignment = Alignment.CenterVertically,
         horizontalArrangement = Arrangement.SpaceBetween,
         modifier = Modifier.fillMaxWidth()) {
           Text(text = association.name, style = AppTypography.bodyMedium)
+          // Role Badge
+          if (userRole != null) {
+            Box(
+                modifier =
+                    Modifier.padding(start = 8.dp)
+                        .background(color = Color(userRole.color), shape = RoundedCornerShape(8.dp))
+                        .padding(horizontal = 8.dp, vertical = 4.dp)) {
+                  Text(
+                      text = userRole.displayName,
+                      style = AppTypography.bodySmall,
+                      color = Color.White,
```

Comment:
As the user role is present in multiple composable, I think taht it would be beneficial to move it as a separate composable. This would avoid code duplication.

Reply:
Good !